server/src/main/java/org/apache/druid/client/indexing/ClientCompactionRunnerInfo.java

```diff
@@ -138,13 +141,19 @@ private static CompactionConfigValidationResult compactionConfigSupportedByMSQEn
 
   /**
    * Validate that partitionSpec is either 'dynamic` or 'range'. If 'dynamic', ensure 'maxTotalRows' is null. If range
-   * ensure all partition columns are of string type.
+   * ensure all partition columns are of single-valued string type .
    */
   public static CompactionConfigValidationResult validatePartitionsSpecForMSQ(
       @Nullable PartitionsSpec partitionsSpec,
-      @Nullable List<DimensionSchema> dimensionSchemas
+      @Nullable List<DimensionSchema> dimensionSchemas,
+      @Nullable Set<String> multiValuedDimensions
   )
   {
+    if (partitionsSpec == null) {
+      return CompactionConfigValidationResult.failure(
+          "MSQ: tuningConfig.partitionsSpec must be set"
```

Comment:
Why do we need this? I don't think specifying `partitionsSpec` is a requirement in native engine?

Reply:
In native, a default spec is applied by the index task based on certain field values: https://github.com/apache/druid/blob/c2149d59a7da1ac4cbb02aaa63a52cb326306dc5/indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java#L1247

This is not the case with MSQ and the task fails. 

In the web-ui for compaction, partitionsSpec is populated with some value by default.